benchbuild/environments/service_layer/unit_of_work.py

```diff
@@ -1,14 +1,176 @@
 import abc
 import typing as tp
 
-from benchbuild.environments.adapters import buildah, podman, repository
-from benchbuild.environments.domain import events, model
+from benchbuild.environments.adapters import buildah, podman
+from benchbuild.environments.domain import commands, events, model
+
+
+class EventCollector(tp.Protocol):
```

Comment:
Before python version 3.8, `Protocol` is defined in `typing_extensions` and not in `typing`.